### ISSUE TITLE
[Question] Alter table drop partition on cluster errors

### ISSUE THREAD

**Issue by igor-sh8** [yandex/ClickHouse#3520]
Hello. 
We have a cluster of 6 nodes (3 shards). Replication configs include login/pass authorization. 
```
......
        <shard>
                <replica>
                    <host>serever1</host>
                    <port>9000</port>
                    <user>user1</user>
                    <password>pass1</password>
                </replica>
                <replica>
                    <host>serever11</host>
                    <port>9001</port>
                    <user>user1</user>
                    <password>pass1</password>
                </replica>
            </shard>
........
```
For insert, select, create queries - all works fine 
But when we run a query like:  
`ALTER TABLE table1 on cluster cluster1 DROP PARTITION '2018-10-01';`
sometimes we got such exception:
```
┌─host───────┬─port─┬─status─┬─error─┬─num_hosts_remaining─┬─num_hosts_active─┐
│ server21 │ 9002 │      0 │       │                   5 │                2 │
│ server3  │ 9000 │      0 │       │                   4 │                2 │
│ server31 │ 9003 │      0 │       │                   3 │                2 │
│ server2  │ 9000 │      0 │       │                   2 │                2 │
└────────────┴──────┴────────┴───────┴─────────────────────┴──────────────────┘
┌─host───────┬─port─┬─status─┬─error──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─num_hosts_remaining─┬─num_hosts_active─┐
│ server11 │ 9001 │    192 │ Code: 192, e.displayText() = DB::Exception: Received from server1:9000, 10.1.0.25. DB::Exception: Unknown user default., e.what() = DB::Exception │                   1 │                1 │
└────────────┴──────┴────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────────┴──────────────────┘
┌─host──────┬─port─┬─status─┬─error─┬─num_hosts_remaining─┬─num_hosts_active─┐
│ server1 │ 9000 │      0 │       │                   0 │                0 │
└───────────┴──────┴────────┴───────┴─────────────────────┴──────────────────┘
```
The main word is sometimes. 
We can run such a query several times and get the same exception, but at some moment a query may works as it should.

Maybe someone has already encountered this?
I would appreciate any tips.

**Comment by 4ertus2** [comment]
It looks like you trying to make operation via user 'default.' not 'default'. There's a strange dot at the end of name. Look at your users.xml files on all the nodes. Is there any <default.> instead of <default>?

Other place where you may look at is connection messages, such as:
```
<Debug> TCPHandler: Connected ClickHouse client version 18.14.0, revision: 54409, user: default.
```
Is there only one dot in all such messages?

**Comment by igor-sh8** [comment]
Hi. 
There was no user "default." or "default" in configs. At the same time, everything was fine in the logs too. 
As a fix I just created user "default" on all nodes in the cluster with strict network limitation (only for cluster nodes). And It seems helps us.

**Comment by vglocus** [comment]
I run a 20 node replicated cluster (10 shards * 2 replicas). Cluster is configured with internal_replication and using user "distributed" for inter-cluster communication.  

I have encountered this same issue both using distributed queries (ON CLUSTER) and local (single Replica Pair). 

Either `ALTER TABLE "foo"."bar_local" ON CLUSTER "my-cluster" DROP PARTITION '201901'` or `ALTER TABLE "foo"."bar_local" DROP PARTITION '201901'` occationally throw the following error:

```Code: 195. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: Received from <replica_host_name>:9000, <replica_host_ip>. DB::Exception: User default is not allowed to connect from address ::ffff:<local_host_external_ip>.```

The user "default" is indeed restricted not work for this IP address, but the default user should not be used anywhere here. I run the command from clickhouse client as the user "local" and cluster is configured to use the user "distributed" internally. This configuration is validated on both hosts.

Running the same command from the replica host there is no exception indicating that this is not a bi-directional relationship.

**Comment by filimonov** [comment]
It's looks like a duplicate of https://github.com/yandex/ClickHouse/issues/1861

It was fixed in 19.1
